com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/LensFlareMergeOcclusionDataDriven.compute

```diff
@@ -0,0 +1,47 @@
+// Definitions
+//--------------------------------------------------------------------------------------------------
+
+// #pragma enable_d3d11_debug_symbols
+#pragma only_renderers d3d11 playstation xboxone xboxseries vulkan metal switch
+
+#pragma kernel MainCS
+
+//#pragma multi_compile _ SSR_APPROX
+
+//--------------------------------------------------------------------------------------------------
+// Included headers
+//--------------------------------------------------------------------------------------------------
+
+#include "Packages/com.unity.render-pipelines.core/ShaderLibrary/Common.hlsl"
+#include "Packages/com.unity.render-pipelines.core/ShaderLibrary/Packing.hlsl"
+#include "Packages/com.unity.render-pipelines.high-definition/Runtime/ShaderLibrary/ShaderVariables.hlsl"
+#include "Packages/com.unity.render-pipelines.high-definition/Runtime/Lighting/ScreenSpaceLighting/ScreenSpaceLighting.hlsl"
+#include "Packages/com.unity.render-pipelines.high-definition/Runtime/Lighting/ScreenSpaceLighting/ShaderVariablesScreenSpaceReflection.cs.hlsl"
+#include "Packages/com.unity.render-pipelines.high-definition/Runtime/Material/Builtin/BuiltinData.hlsl"
+
+#ifdef DEBUG_DISPLAY
+#include "Packages/com.unity.render-pipelines.core/ShaderLibrary/Debug.hlsl"
+#include "Packages/com.unity.render-pipelines.high-definition/Runtime/Debug/DebugDisplay.hlsl"
+#endif
+
+RW_TEXTURE2D_X(float, _LensFlareOcclusion);
+
+[numthreads(8, 1, 1)]
+void MainCS(uint3 dispatchThreadId : SV_DispatchThreadID)
```

Comment:
@soufianekhiat while this is fine for now, I wanna challenge you:
You have an entire wave (64 threads) that you could exploit, you could dispatch a 8x8 (64 threads) and in the y coordinate load your history into lds.
You can then do a parallel reduction or interlock min on a final value and avoid tapping 8 times the history resource :)

Just fun programming if you are up to it! Please dont consider this as a blocker

Reply:
Updated with 3 barriers & LDS